soccer/src/soccer/ui/main_window.cpp

```diff
@@ -1146,3 +1170,154 @@ void MainWindow::updateDebugLayers(const LogFrame& frame) {
+    onPositionDropdownChanged(0, value);
+}
+
+void MainWindow::on_robotPosition_1_currentIndexChanged(int value) {
+    onPositionDropdownChanged(1, value);
+}
+
+void MainWindow::on_robotPosition_2_currentIndexChanged(int value) {
+    onPositionDropdownChanged(2, value);
+}
+
+void MainWindow::on_robotPosition_3_currentIndexChanged(int value) {
+    onPositionDropdownChanged(3, value);
+}
+
+void MainWindow::on_robotPosition_4_currentIndexChanged(int value) {
+    onPositionDropdownChanged(4, value);
+}
+
+void MainWindow::on_robotPosition_5_currentIndexChanged(int value) {
+    onPositionDropdownChanged(5, value);
+}
+
+void MainWindow::on_robotPosition_6_currentIndexChanged(int value) {
+    onPositionDropdownChanged(6, value);
+}
+
+void MainWindow::on_robotPosition_7_currentIndexChanged(int value) {
+    onPositionDropdownChanged(7, value);
+}
+
+void MainWindow::on_robotPosition_8_currentIndexChanged(int value) {
+    onPositionDropdownChanged(8, value);
+}
+
+void MainWindow::on_robotPosition_9_currentIndexChanged(int value) {
+    onPositionDropdownChanged(9, value);
+}
+
+void MainWindow::on_robotPosition_10_currentIndexChanged(int value) {
+    onPositionDropdownChanged(10, value);
+}
+
+void MainWindow::on_robotPosition_11_currentIndexChanged(int value) {
+    onPositionDropdownChanged(11, value);
+}
+
+void MainWindow::on_robotPosition_12_currentIndexChanged(int value) {
+    onPositionDropdownChanged(12, value);
+}
+
+void MainWindow::on_robotPosition_13_currentIndexChanged(int value) {
+    onPositionDropdownChanged(13, value);
+}
+
+void MainWindow::on_robotPosition_14_currentIndexChanged(int value) {
+    onPositionDropdownChanged(14, value);
+}
+
+void MainWindow::on_robotPosition_15_currentIndexChanged(int value) {
+    onPositionDropdownChanged(15, value);
+}
+
+void MainWindow::onResetButtonClicked(int robot) {
+    rj_msgs::msg::OverridePosition message;
+    message.robot_id = robot;
+    message.overriding_position = 0;
+
+    test_play_pub_->publish(message);
+    position_reset_buttons.at(robot)->setEnabled(false);
+    robot_pos_selectors.at(robot)->setCurrentIndex(0);
+}
+
+void MainWindow::onPositionDropdownChanged(int robot, int position_number) {
+    // This is a convoluted scheme to figure out who the goalie is
+    // May not be necessary later
+    QString goalieNum = _ui.goalieID->currentText();
+    bool goalieNumIsInt{false};
+    int goalieInt = goalieNum.toInt(&goalieNumIsInt);
+    if (!goalieNumIsInt || goalieInt != robot) {
+        rj_msgs::msg::OverridePosition message;
+        message.robot_id = robot;
+        message.overriding_position = position_number;
+
+        test_play_pub_->publish(message);
+        if (position_number != 0) {
+            position_reset_buttons.at(robot)->setEnabled(true);
+        } else {
+            position_reset_buttons.at(robot)->setEnabled(false);
+        }
+    }
+}
+
+void MainWindow::setGoalieDropdown(int robot) {
```

Comment:
Is it possible to somehow store the value we set the Goalie to here so we don't need the scheme above to find out who the goalie is? And by default assume it is robot 0 if we have not set anything since robot 0 is the default set to goalie?